designer/report-designer-extension-pentaho/src/main/java/org/pentaho/reporting/designer/extensions/pentaho/repository/actions/PublishToServerTask.java

```diff
@@ -78,8 +81,12 @@ public void run() {
 
       reportDesignerContext.getActiveContext().getAuthenticationStore().add( loginData, storeUpdates );
 
+      //populate all properties from file which loaded in report designer before publish it to server
+      Optional<Properties> fileProperties = Optional.of( new Properties() );
```

Comment:
Optionals with collections is not needed. Your code does not rely on whether the properties collection is given or no, it treats the properties-optional as if there is always such an collection (by creating one on the fly if needed). So make it easy and simply pass the collection itself. 


For the legacy code